core/src/main/scala/org/ensime/indexer/DescriptorParser.scala

```diff
@@ -2,13 +2,47 @@
 // Licence: http://www.gnu.org/licenses/gpl-3.0.en.html
 package org.ensime.indexer
 
+import org.ensime.indexer.ClassName._
 import org.parboiled2.Parser
 import org.parboiled2.Rule1
 import org.parboiled2._
 
 import scala.annotation.switch
 import scala.util.{ Failure, Success }
 
+trait ClassParser extends Parser {
```

Comment:
some things in here could be shared, but not everything. Try to use an imported `object` in preference to a mix in trait if you can.